client/state/data-layer/wpcom-http/actions.js

```diff
@@ -3,10 +3,75 @@
  */
 import { WPCOM_HTTP_REQUEST } from 'state/action-types';
 
+export const apiVersionPattern = /^v(?:(\d+)\.)?(\d+)$/;
+export const apiNamespacePattern = /^([a-z]+)\/?v(\d+)$/;
+
+const noVersionAndNamespaceUndefined = ( { apiNamespace, apiVersion } ) => {
+	if ( typeof apiNamespace === 'undefined' && typeof apiVersion === 'undefined' ) {
+		throw new TypeError(
+			'One of `apiNamespace` and `apiVersion` must be defined (see wpcom.js/README.md)'
+		);
+	}
+};
+
+const noVersionAndNamespaceConflicts = ( { apiNamespace, apiVersion } ) => {
+	if ( apiNamespace && apiVersion ) {
+		throw new TypeError(
+			'Cannot simultaneously define `apiNamespace` and `apiVersion` (see wpcom.js/README.md)'
+		);
+	}
+};
+
+const hasViableVersion = ( { apiVersion, apiNamespace } ) => {
+	if ( ! apiVersion && apiNamespace ) {
+		return;
```

Comment:
this return statement is awfully ambiguous. the function has a predicate name `has_` but returns `undefined` on every invocation. does that mean is _does_ have a viable version? `undefined` is falsely so if we were to see `if ( hasViableVersion( … ) )` then that conditional would always return false.

what if we simply returned `true` in all cases where we don't `throw`